packages/flutter_sprites/lib/src/sound.dart

```diff
@@ -1,139 +1,111 @@
 part of flutter_sprites;
 
-// TODO: The sound effects should probably use Android's SoundPool instead of
-// MediaPlayer as it is more efficient and flexible for playing back sound effects
-
-typedef void SoundEffectStreamCallback(SoundEffectStream stream);
-
+/// An audio asset loaded by the SoundEffectPlayer.
 class SoundEffect {
-  SoundEffect(this._pipeFuture);
-
-  // TODO: Remove load method from SoundEffect
-  Future load() async {
-    _data = await _pipeFuture;
-  }
+  int _soundId;
 
-  Future<MojoDataPipeConsumer> _pipeFuture;
-  MojoDataPipeConsumer _data;
+  SoundEffect(this._soundId);
```

Comment:
constructor first, please. see the style guide at https://github.com/flutter/engine/blob/master/sky/specs/style-guide.md